clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -2136,15 +2136,12 @@ Address CodeGenFunction::EmitIntelFPGAFieldAnnotations(SourceLocation Location,
                                                        StringRef AnnotStr) {
   llvm::Value *V = Addr.getPointer();
   llvm::Type *VTy = V->getType();
+  assert(isa<llvm::PointerType>(VTy) &&
+      "second argument of llvm.ptr.annotation should be pointer");
   llvm::Function *F =
-      CGM.getIntrinsic(llvm::Intrinsic::ptr_annotation, CGM.Int8PtrTy);
-  // FIXME Always emit the cast inst so we can differentiate between
-  // annotation on the first field of a struct and annotation on the struct
-  // itself.
-  if (VTy != CGM.Int8PtrTy)
-    V = Builder.CreateBitCast(V, CGM.Int8PtrTy);
+      CGM.getIntrinsic(llvm::Intrinsic::ptr_annotation, VTy);
   V = EmitAnnotationCall(F, V, AnnotStr, Location);
-  V = Builder.CreateBitCast(V, VTy);
+  V = Builder.CreatePointerBitCastOrAddrSpaceCast(V, VTy);
```

Comment:
Annotation intrinsic returns the same type that was passed in, so you don't need this line now.

Reply:
Indeed, thanks!